contracts/Inbox.sol

```diff
@@ -16,6 +40,8 @@ import {ISemver} from "./interfaces/ISemVer.sol";
 contract Inbox is IInbox, Ownable {
     using TypeCasts for address;
 
+    uint256 public constant MAX_BATCH_SIZE = 10;
```

Comment:
im pretty sure all the inbox changes are fine, but another reason why its kind of nice to have the work streams separate is that im reviewing code that's already merged into a different feature branch, but i have to look at it again as though its a new thing, because you may have made some changes to it, and github has no way of separating code merged elsewhere from fresh code here. I totally get what you're saying too, im trying to think of a better way here, not even sure if there is one. maybe a submodule situation?

Reply:
for the time being, I'd wanna merge HYPERPROVER into main (which should be a pretty simple approve - hyperprover code is literally live right now), then merge main into beta, and merge beta into this. That should take out the noise from most of the approved hyperprover stuff